pages/settings/script.js

```diff
@@ -1,84 +1,207 @@
 
-const user = getUserSession();
+const user = getUserSession()
 
 function showUserInfo() {
-  if (user) {
-    document.getElementById("name").setAttribute("value", user?.name);
-    document.getElementById("email").setAttribute("value", user?.email);
-  } else {
-    myProfile.style.display = "none";
-    deleteAccount.style.display = "none";
-  }
+	if (user) {
+		document.getElementById('name').setAttribute('value', user?.name)
+		document.getElementById('email').setAttribute('value', user?.email)
+	} else {
+		myProfile.style.display = 'none'
+		deleteAccount.style.display = 'none'
+	}
 }
 
-showUserInfo();
+showUserInfo()
 
 // Change user information
-let userName = user?.name;
-let userEmail = user?.email;
-
-saveProfileInfoBtn.addEventListener("click", () => {
-  userName = document.getElementById("name").value;
-  user.name = userName;
-  userEmail = document.getElementById("email").value;
-  user.email = userEmail;
-  setUserSession(user);
-  showProfilePic();
-});
+let userName = user?.name
+let userEmail = user?.email
+let userID = user?.id
+
+saveProfileInfoBtn.addEventListener('click', (e) => {
+	e.preventDefault()
+	userName = document.getElementById('name').value
+	userEmail = document.getElementById('email').value
+	user.name = userName
+	user.email = userEmail
+	setUserSession(user)
+	// showProfilePic()
+	saveUserInfo()
+	saveProfilePicture()
+})
+
+// Save user information
+async function saveUserInfo() {
+	try {
+		const name = document.getElementById('name').value
+
+		const response = await fetch(`${API_URL}/user?id=${userID}`, {
+			method: 'PUT',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({
+				name,
+			}),
+		})
+		const result = await response.json()
+		console.log('user name updated succesfully', result)
+	} catch (error) {
+		console.log('user name error', error)
+	}
+}
 
 // Change password
 function togglePwModal() {
-  const pwModal = changePwPopup.style;
-  pwModal.display = pwModal.display === "block" ? "none" : "block";
+	const pwModal = changePwPopup.style
+	pwModal.display = pwModal.display === 'block' ? 'none' : 'block'
 }
 
-changePwBtn.addEventListener("click", togglePwModal);
+changePwBtn.addEventListener('click', togglePwModal)
 
-changePwSaveBtn.addEventListener("click", togglePwModal);
+changePwSaveBtn.addEventListener('click', togglePwModal)
 
 // Profile photo
-
 function showProfilePic() {
-  profilePhoto.setAttribute("src", user?.photo);
+	if (user && user.photo) {
+		profilePhoto.setAttribute('src', user?.photo)
+	} else {
+		profilePhoto.setAttribute('src', '#')
+	}
 }
 
-showProfilePic();
-
-const dropArea = document.getElementById("fileDropWrapper");
-
-dropArea.addEventListener("dragover", (e) => {
-  e.preventDefault();
-});
+showProfilePic()
 
-dropArea.addEventListener("drop", (e) => {
-  e.preventDefault();
+const dropArea = document.getElementById('dropArea')
+const inputFile = document.getElementById('inputImage')
+var picture
```

Comment:
Is there a reason why have a `var` variable and not a `let`.
Also is a good idea to have your global variables at the top of the file

Reply:
fixed in commit 3f39481